src/signals-unix.c

```diff
@@ -789,21 +796,137 @@ void trigger_profile_peek(void)
     jl_safe_printf("\n======================================================================================\n");
     jl_safe_printf("Information request received. A stacktrace will print followed by a %.1f second profile\n", profile_peek_duration);
     jl_safe_printf("======================================================================================\n");
-    if (bt_size_max == 0){
+    if (profile_bt_size_max == 0){
         // If the buffer hasn't been initialized, initialize with default size
         // Keep these values synchronized with Profile.default_init()
         if (jl_profile_init(10000000, 1000000) == -1) {
             jl_safe_printf("ERROR: could not initialize the profile buffer");
             return;
         }
     }
-    bt_size_cur = 0; // clear profile buffer
-    if (jl_profile_start_timer() < 0)
+    profile_bt_size_cur = 0; // clear profile buffer
+    if (jl_profile_start_timer(0) < 0)
         jl_safe_printf("ERROR: Could not start profile timer\n");
     else
         profile_autostop_time = jl_hrtime() + (profile_peek_duration * 1e9);
 }
 
+void jl_profile_task_unix(void)
+{
+    if (jl_profile_is_buffer_full()) {
+        // Buffer full: Delete the timer
+        jl_profile_stop_timer();
+        return;
+    }
+
+    // Get a random task for which we know that we won't
+    // drop a sample in the profiler
+    // TODO(Diogo, Nick): should we put an upper bound on the number of iterations?
+    arraylist_t *tasks = jl_get_all_tasks_arraylist();
```

Comment:
Can't you lift this to something above. I suspect you won't be getting only one sample.

Reply:
The number of tasks may change during the interval between two samples though.

In any case, seems like hoisting this won't bring significant performance benefit: https://github.com/RelationalAI/julia/pull/186#discussion_r1775177192 -- it was only spending 1.5% of time in this function getting all tasks.

Reply:
Ok seems reasonable then